### PR TITLE
Ali/reading collection/index

### PR DESCRIPTION
**Pull request by mintona**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Adds a Reading Collection model to the DB
- Adds joins table called Reading Collection Articles Joins table for many to many between reading collections and articles
- Set-up all relationships according to the chore card tagged below 

## Related Tickets & Documents
#15 
#14 
## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

* There are 2 failing tests in `services/users/delete_spec.rb` that need to be de-bugged, due to our lack of answers around what exactly the tests are doing. 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed
- [x] will need to be added

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/reading_list_items_controller.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 class ReadingListItemsController < ApplicationController
   def index
     @reading_list_items_index = true
+    @reading_collections = ReadingCollection.where(user_id: session_current_user_id).to_a
+    # Article.joins(:taggings).joins(:tags).select("articles.*").where("tags.name = 'javascript'")
```
Comment (joel-oe-lacey) [inline app/controllers/reading_list_items_controller.rb:5]: Is this laying out functionality to generate a temporary collection for testing, or pseudocode of how we might generate article per collection later?

Comment (joel-oe-lacey) [reply app/controllers/reading_list_items_controller.rb:5]: Not applicable for this moment, just a note to touch on in standup.

**Review by joel-oe-lacey — COMMENTED** [review]
Looks good to my understanding of things, cool seeing the flow of all these pieces meshing. I made some changes to the readingList view, but just a reference point change that shouldn't conflict with anything. Shipping it!

File: app/controllers/reading_list_items_controller.rb (hunk 1 above)
Comment (mintona) [reply app/controllers/reading_list_items_controller.rb:5]: Yup, it's more like a "note" for later when we add the ability to find articles of a certain tag.

**Pull request merged**